### PR TITLE
glTF 1.1 draft

### PR DESCRIPTION
**Pull request by lexaknyazev**
* Overview of spec changes: #605
* Direct link to 1.1 spec: https://github.com/lexaknyazev/glTF/tree/master-test/specification/1.1
* Direct link to branch: https://github.com/lexaknyazev/glTF/tree/master-test

---

This PR consists of
- rewritten git history of `1.0.1` branch with changed paths to `specification/1.1` directory;
- updated JSON snippets inside README.md - now each of them is a correct JSON;
- fixes for typos throughout spec, schemas, and examples;
- JSON schemas are updated to v4;
- all recent updates from `master` branch.

---

**Specification TODOs**

- [ ] Update Overview / Reference Card (#761)
- [x] Tighten up binary data layout, issue isn't finished yet (#749)
- [ ] Skinning update (https://github.com/KhronosGroup/glTF/issues/723#issuecomment-254674004)

**Implementation TODOs**

- [ ] Cesium, https://github.com/AnalyticalGraphicsInc/cesium/issues/4009
- [ ] three.js, https://github.com/mrdoob/three.js/issues/10257
- [ ] BabylonJS
- [ ] COLLADA2GLTF, https://github.com/KhronosGroup/COLLADA2GLTF/issues/3
- [ ] obj2gltf, https://github.com/AnalyticalGraphicsInc/obj2gltf/issues/35
- [ ] gltf-pipeline, https://github.com/AnalyticalGraphicsInc/gltf-pipeline/issues/188

**Other TODOs**
- [ ] [sample models](https://github.com/KhronosGroup/glTF-Sample-Models) for glTF 1.1
- [ ] Close issues labeled [1.1](https://github.com/KhronosGroup/glTF/labels/1.1), all of which should also be labeled `resolved`

### CODE REVIEW

**Comment by pjcozzi** [comment]
Awesome, @lexaknyazev!

* Overview of spec changes: #605
* Direct link to 1.1 spec: https://github.com/lexaknyazev/glTF/tree/master-test/specification/1.1
* Direct link to branch: https://github.com/lexaknyazev/glTF/tree/master-test

**Comment by pjcozzi** [comment]
I made a few trivial tweaks to the spec links.

**Comment by pjcozzi** [comment]
@lasalvavida I added an **Implementation TODO** section to the top of this pull request for implementations I think it would be good to have before merging this.  Is there anything else we should create a tasklist for that needs to be done before merging this?  Is the validation completely in-sync with the spec?

**Comment by pjcozzi** [comment]
I don't know if there is an easier way to do this with GitHub given the history changes, but copying and pasting the 1.0 and 1.1 spec into https://www.diffchecker.com/diff has worked really well for me.

**Comment by lexaknyazev** [comment]
> Is there anything else we should create a tasklist for that needs to be done before merging this?

Updated top message with TODO. I'd merge it anyway to bring everyone's attention to updated spec (we can state on the main repo page that it's a draft).

> Is the validation completely in-sync with the spec?

Command-line version - yes; web is a bit outdated, I'll rebuild it soon. Also we need to finish those two open spec issues.

**Comment by pjcozzi** [comment]
> I'd merge it anyway to bring everyone's attention to updated spec (we can state on the main repo page that it's a draft).

We can discuss today, but I would rather keep master stable and let users refer to this branch while we get a few implementations together.

**Comment by pjcozzi** [comment]
@lexaknyazev is COLLADA2GLTF updated for glTF 1.1?

**Comment by lexaknyazev** [comment]
> @lexaknyazev is COLLADA2GLTF updated for glTF 1.1?

https://github.com/KhronosGroup/COLLADA2GLTF/issues/3

**Comment by pjcozzi** [comment]
>> @lexaknyazev is COLLADA2GLTF updated for glTF 1.1?
> KhronosGroup/COLLADA2GLTF#3

Added to tasklist.

File: specification/1.1/README.md [hunk 1]
```diff
@@ -0,0 +1,3785 @@
...
+where
+- `a` equals `camera.perspective.aspectRatio`;
+- `y` equals `camera.perspective.yfov`;
+- `f` equals `camera.perspective.zfar`;
+- `n` equals `camera.perspective.znear`.
+
+#### Orthographic projection
+<p><img src="figures/ortho.png" /></p>
+where
+- `r` equals `camera.orhographic.xmag`;
```
Comment (lasalvavida) [inline specification/1.1/README.md]: I think this is a typo. Shouldn't this be `orthographic`?

Comment (lexaknyazev) [inline specification/1.1/README.md]: Sure. Fixed with https://github.com/lexaknyazev/glTF/commit/36042431f1145cec6c7c51d62082af0b942c9982.

**Comment by xelatihy** [comment]
Is the changes to use top level arrays as opposed to dictionary already being implemented in the draft spec?

**Comment by pjcozzi** [comment]
The draft spec is not updated yet for the object to array changes.

**Comment by lexaknyazev** [comment]
@pjcozzi 
Please, review https://github.com/KhronosGroup/glTF/pull/784/commits/2113534b2de3fa1c0ed66b14ab4f23d7bf015f46.

**Comment by pjcozzi** [comment]
> @pjcozzi 
Please, review https://github.com/KhronosGroup/glTF/pull/784/commits/2113534b2de3fa1c0ed66b14ab4f23d7bf015f46.

Looks good, except this is vague and maybe misleading:

> While some implementations could handle misaligned data, not following this recommendation may lead to extra data processing, performance degradation, and/or inability to initialize attributes pipeline.

Perhaps remove it.

**Comment by lexaknyazev** [comment]
> Looks good, except this is vague and maybe misleading...

The point was that, depending on driver / API misaligned data could be rendered (because driver will fix that).

**Comment by pjcozzi** [comment]
OK, let's be more precise or remove it.

**Comment by lexaknyazev** [comment]
Continues in #826.

**Pull request closed without merging**